boot/mbed/mcuboot_imgtool.cmake

```diff
@@ -76,6 +66,18 @@ function(mcuboot_generate_initial_image TARGET) # optional 2nd arg: file name
     endif()
     set(SLOT_SIZE_HEX ${CMAKE_MATCH_1})
 
+    get_property(objcopy GLOBAL PROPERTY ELF2BIN)
+
+    if(${IMAGE_TYPE} STREQUAL "initial")
+        set(IMGTOOL_EXTRA_ARGS
+            # this should only be used for binaries you plan on initially flashing to your target at the factory.
+            # It pads the resulting binary to the slot size and adds initialized trailer TLVs. This is not needed
+            # for update binaries.
```

Comment:
this isn't needed for swap modes by default, they should boot without it, it's needed for directXIP

Reply:
Hmm, interesting.  I got this from George's docs, so I guess he was wrong about that?  Does that mean that one can use the same image for the initial flash and for the update?

Reply:
I am not sure if Mr. Beckstein is wrong here. At least as of v1.8.0, if a padded image is downloaded into secondary slot, bootloader will upgrade from it even primary application hasn't called `boot_set_pending()` yet. Could you check what `boot_set_pending()` does? I assume it adds a bunch of TLVs so bootloader can recognize and accept the image in secondary slot.

Reply:
I think what @nordicjm is saying is that we should remove this line for everything and just use the "update" image for all use cases

Reply:
As it's done in zephyr, you build once without confirmation and flash the hex file, there is a bin file equivalent which you use for firmware updates (that you must mark for update using test or confirmed) - this applies to swap modes since they use the flags. DirectXIP does not use any flags, it will just boot the newest image. The exception is directXIP with revert, this requires the confirmed flag even on the initial image as it uses that to determine if the image is valid and to be booted

Reply:
gotcha, removed.